docs/guides/tutorials/rest_apis_with_fastapi.rst

```diff
@@ -1167,16 +1170,403 @@ payload:
+
+    CONFIGURE CURRENT BRANCH SET
+    ext::auth::SMTPConfig::security := 'STARTTLSOrPlainText';
+
+    CONFIGURE CURRENT BRANCH SET
+    ext::auth::SMTPConfig::validate_certs := false;
+
+You can query the database configuration to discover which providers are 
+configured with the following query:
+
+.. code-block:: edgeql
+
+  select cfg::Config.extensions[is ext::auth::AuthConfig].providers {
+      name,
+      [is ext::auth::OAuthProviderConfig].display_name,
+  };
+
+Implementing authentication with FastAPI
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Below, we provide a detailed guide to setting up authentication using FastAPI,
+including both sign-in and sign-up functionalities.
+
+PKCE flow for enhanced security
+-------------------------------
+
+The PKCE (Proof Key for Code Exchange) flow enhances security in server-to-server
+authentication by generating a unique verifier and its corresponding challenge.
+First, your server creates a 32-byte Base64 URL-encoded verifier, stores it in an
+HttpOnly cookie, hashes it with SHA256, and then encodes it to form the challenge.
+
+This implementation ensures enhanced security by preventing token leakage and is
+tailored specifically for server-to-server interactions.
+
+Add the following code to your FastAPI application to generate the PKCE:
+
+.. code-block:: python
+    :caption: app/auth.py
+
+    import secrets
+    import hashlib
+    import base64
+
+    def generate_pkce():
+        verifier = secrets.token_urlsafe(32)
+        challenge = hashlib.sha256(verifier.encode()).digest()
+        challenge_base64 = base64.urlsafe_b64encode(challenge).decode('utf-8').rstrip('=')
+        return verifier, challenge_base64
+
+User registration and authentication
+------------------------------------
+
+Next, we're going to create endpoints in FastAPI to handle user registration
+(sign-up) and user login (sign-in):
+
+**Sign-up endpoint**
+
+.. code-block:: python
+    :caption: app/auth.py
+
+    from fastapi import APIRouter, HTTPException, Request
+    from fastapi.responses import JSONResponse
+    import httpx
+
+    router = APIRouter()
+    
+    # Value should be:
+    # {protocol}://${host}:${port}/branch/${branch}/ext/auth/
+    EDGEDB_AUTH_BASE_URL = os.getenv('EDGEDB_AUTH_BASE_URL')
+
+    @router.post("/auth/signup")
+    async def handle_signup(request: Request):
+        body = await request.json()
+        email = body.get("email")
+        password = body.get("password")
+        name = body.get("name")
+
+        if not email or not password or not name:
+            raise HTTPException(status_code=400, detail="Missing email, password, or name.")
+
+        verifier, challenge = generate_pkce()
+        register_url = f"{EDGEDB_AUTH_BASE_URL}/register"
+        response = httpx.post(register_url, json={
+            "challenge": challenge,
+            "email": email,
+            "password": password,
+            "provider": provider,
```

Comment:
Where does `provider` come from here? I think we can hardcode this as `builtin::local_emailpassword` since this endpoint just deals with email+password sign up.